### PR TITLE
list-ops: add hint of using static methods per #378

### LINKED ISSUE #378: list-ops: include hints to explain why methods are static
As per #177, we prefer using instance methods over class methods (see discussion in that issue for rationale).

However, there _are_ conditions in which it is absolutely appropriate for a function to be `static`.  A common situation is because the function is effectively an extension of an existing type — you are defining behavior you wished were defined on a given type.

Since classes in Java are closed for modification (i.e. you cannot add members to a class outside its definition like you can in other languages like Ruby or JavaScript), you cannot add the new behavior to the class, directly.  The idiomatic solution in this case is to write a utility method.

Collections of these kinds of methods are often referred to as "utility classes".  Examples of such classes from within the JRE include [`Arrays`](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html) and [`Collections`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html).

In the case of the exercise `list-ops`, we "wish" that [`List`](https://docs.oracle.com/javase/8/docs/api/java/util/List.html) had `map()`, `reduce()`, `filter()`, etc. methods on it.  It doesn't, so we're including them here.

Please include a `HINT.md` file in `list-ops` that points out that these methods are static and capture the description above, in your own words as to why this class has this shaped API.

### PR DESCRIPTION
**Pull request by m-a-ge**
Hopefully resolves #378 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/list-ops/.meta/hints.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 ## Hints
 
+In Java it's a best pracite to use instance methods over class methods. However, there are conditions in which it is absolutely appropriate for a function to be `static`. Since classes in Java are closed for modification (i.e. you cannot add members to a class outside its definition like you can in other languages like Ruby or JavaScript), you cannot add new behavior to the class, directly. What to do if you still want to define behavior for a given type? The idiomatic solution in this case is to write a utility method.
```
Comment (FridaTveit) [inline exercises/list-ops/.meta/hints.md]: Should be "practice" :) And I don't think there should be a comma before "directly"? I'm not an expert on comma usage though so feel free to tell me if I'm wrong :P

Comment (FridaTveit) [reply exercises/list-ops/.meta/hints.md]: Also I think it should be "it's considered best practice" instead of "it's a best practice". What do you think?

Comment (m-a-ge) [reply exercises/list-ops/.meta/hints.md]: @FridaTveit you are right, I'll fixes those later today.
Thanks for pointing out 👍

**Comment by m-a-ge** [comment]
@FridaTveit I've corrected those annoying mistakes

**Pull request merged**